File: 임준형/kotlin-grammer/src/lec02/java/Person.java
package lec02.java;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class Person{
    String name;
    int age;

    public Person(String name) {
        this.name = name;
    }

    public Person(final String name, final int age) {
        this.name = name;
        this.age = age;
    }

    //    @Nullable
    @NotNull
    public String getName() {
        return name;
    }

    public int getAge() {
        return age;
    }
}

File: 김아린/KotlinStudy/src/main/kotlin/Person.kt
import java.lang.IllegalArgumentException

fun main() {
    Person()
}

class Person constructor(
    val name: String = "김아린",
    var age: Int = 1
) {
    val uppercaseName: String
        get() = this.name.uppercase()

    init {
        if (age < 0) {
            throw IllegalArgumentException("나이는 ${age}일 수 없습니다")
        }
        println("초기화")
    }

}